### PR TITLE
misc: fix dangling pointer in printf_buffer

### PR DESCRIPTION
**Pull request by RABijl**
also includes fixing uint to int conversion
warning and some other clang tidy warnings.

### CODE REVIEW

**Comment by pjaaskel** [comment]
Flakiness occurs here with the LLVM 17 bot: `conformance_main_images_cl_fill_images`. Some image tests failed before me as well sporadically.

File: lib/CL/devices/printf_buffer.c [hunk 1]
```diff
@@ -377,7 +371,7 @@ __pocl_printf_format_full (param_t *p, char *buffer, uint32_t buffer_size)
                             "printf error: vector-length is zero\n");
                           return -1;
                         }
-                      if (vector_length > (UINT32_MAX - 9) / 10)
+                      if (vector_length < 0)
```
Comment (franz) [inline lib/CL/devices/printf_buffer.c:374]: Signed integer overflow is undefined behaviour. In this case though, we can just change the condition to: `vector_length > 16` - there are no valid larger vector lengths (and AFAIK no extensions with larger vectors).

**Review by franz — CHANGES_REQUESTED** [review]
small nitpick

File: lib/CL/devices/printf_buffer.c (hunk 1 above)
Comment (RABijl) [reply lib/CL/devices/printf_buffer.c]: fixed.

**Review by franz — APPROVED** [review]

**Pull request merged**